### PR TITLE
[ASUYTP] Write manage.py command for update first name, last name, and name field

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
Write manage.py command for update first name, last name, and the name field in UserProfile

**Description:** Write manage.py command for update first name, last name, and the name field in UserProfile

**Youtrack:** https://youtrack.raccoongang.com/issue/ASUYT-115

**Example:**
```
../manage.py lms --settings=aws  update_user_fullname
```

### CODE REVIEW

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    """
+    help = "Update user's fullname"
+ 
+    def handle(self, *args, **options):
+       """
+       Realisation the command to update User's first name, last name  and the name field in UserProfile.
+       """
+       self.stdout.write("Start to update User's first name and last name  and the name field in UserProfile")
+       user_social_auth_list = UserSocialAuth.objects.filter(
+           Q(user__first_name__icontains='') | Q(user__last_name__icontains='') | Q(user__profile__name__icontains='')
```
Comment (idegtiarov) [inline lms/djangoapps/ytp/management/commands/update_user_fullname.py]: Do we need `icontains` for filtering empty strings? It seems  user__first_name='' is fair enough or I miss something?

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+from logging import getLogger
+
+import requests
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import Q
+from social_django.models import UserSocialAuth
+
+log = getLogger(__name__)
```
Comment (idegtiarov) [inline lms/djangoapps/ytp/management/commands/update_user_fullname.py]: log is not a good approach for the manage command, `stdout.write` suits better.

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+           auth_url = '{}/api/v1/user/login.json'.format(settings.FEATURES['DRUPAL_PRIVIDER_URL'])
+           user_info_url = '{}/api/v1/user/{{}}.json'.format(settings.FEATURES['DRUPAL_PRIVIDER_URL'])
+           session = requests.session()
+           session.headers['Content-type'] = 'application/json'
+    
+           try:
+               r = session.post(token_url)
+               csrf_token = None
+               if r.ok:
+                   log.info('Get the API token')
```
Comment (idegtiarov) [inline lms/djangoapps/ytp/management/commands/update_user_fullname.py]: ditto

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                   csrf_token = r.json().get('token')
+        
+               if csrf_token:
+                   log.info('Post API auth data')
```
Comment (idegtiarov) [inline lms/djangoapps/ytp/management/commands/update_user_fullname.py]: ditto

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                               social_auth_user.user.profile.name = full_name
+                               social_auth_user.user.profile.save()
+                               full_name_list = full_name.split()
+                               fname, lname = full_name_list and (full_name_list[0], ' '.join(full_name_list[1:])) or ('', '')
+                               social_auth_user.user.first_name = fname
+                               social_auth_user.user.last_name = lname
+                               social_auth_user.user.save()
+                               self.stdout.write("User {} updated".format(full_name))
+           except Exception as e:
+               log.error('Exception %s', e)
```
Comment (idegtiarov) [inline lms/djangoapps/ytp/management/commands/update_user_fullname.py]: ditto

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+               raise e
```
Comment (idegtiarov) [inline lms/djangoapps/ytp/management/commands/update_user_fullname.py]: Why do we need to raise an exception? On which step are we expecting an error? If it happens on the user's data updating I would expect that we log some msg to the console and continue our task.

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+class Command(BaseCommand):
+    """
+    Update a User's first name, last name  and the name field in UserProfile from remote service .
+    """
+    help = "Update user's fullname"
+ 
+    def handle(self, *args, **options):
+       """
+       Realisation the command to update User's first name, last name  and the name field in UserProfile.
```
[the remark this answers is not in the document]
Comment (AndreyLykhoman) [reply lms/djangoapps/ytp/management/commands/update_user_fullname.py]: I rewrote doc string.

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+       """
+       self.stdout.write("Start to update User's first name and last name  and the name field in UserProfile")
```
[the remark this answers is not in the document]
Comment (AndreyLykhoman) [reply lms/djangoapps/ytp/management/commands/update_user_fullname.py]: I added changes.

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py (hunk 1 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/ytp/management/commands/update_user_fullname.py]: You are right. user__first_name='' id fair enough. I fixed it.

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py (hunk 2 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/ytp/management/commands/update_user_fullname.py]: I deleted log and used stdout.write.

File: lms/djangoapps/ytp/management/commands/update_user_fullname.py (hunk 6 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/ytp/management/commands/update_user_fullname.py]: I deleted raise exception. And added log some msg if we have some error on the user's data updating.

**Pull request merged**